extensions/src/main/java/org/pentaho/platform/plugin/services/importer/SolutionImportHandler.java

```diff
@@ -991,10 +992,36 @@ private boolean processZip( InputStream inputStream ) {
           }
 
           if ( !fileService.isValidFileName( decodedEntryName ) ) {
-            getLogger().error( Messages.getInstance().getString( "DefaultImportHandler.ERROR_0011_INVALID_FILE_NAME", decodedEntryName ) );
-            throw new PlatformImportException(
-                Messages.getInstance().getString( "DefaultImportHandler.ERROR_0011_INVALID_FILE_NAME",
-                    entryName ), PlatformImportException.PUBLISH_PROHIBITED_SYMBOLS_ERROR );
+            // Attempt retry with sanitized filename
+            String sanitizedName = decodedEntryName.replaceAll( INVALID_FILENAME_CHARS_REGEX, "_" );
+            if ( fileService.isValidFileName( sanitizedName ) ) {
+              getLogger().warn( Messages.getInstance().getString( "SolutionImportHandler.RetryImport", sanitizedName ) );
+              File fileSanitized = new File( sanitizedName );
+              // Create temp file for sanitized name and copy contents
+              File tempFileSanitized = File.createTempFile("zip", null);
+              tempFileSanitized.deleteOnExit();
+              try (FileOutputStream fosSanitized = new FileOutputStream(tempFileSanitized)) {
+                IOUtils.copy(zipInputStream, fosSanitized);
+              }
+              RepositoryFile repoFile = new RepositoryFile.Builder( fileSanitized.getName() ).folder( false ).hidden( false ).build();
+              String parentDir = fileSanitized.getParent() == null ? RepositoryFile.SEPARATOR : fileSanitized.getParent() + RepositoryFile.SEPARATOR;
+              IRepositoryFileBundle repoFileBundle = new RepositoryFileBundle( repoFile, null, parentDir, tempFileSanitized, UTF_8, null );
+              if ( EXPORT_MANIFEST_XML_FILE.equals( fileSanitized.getName() ) ) {
+                initializeAclManifest( repoFileBundle );
+              } else {
+                if ( isPerformingRestore ) {
+                  getLogger().debug( "Adding file " + repoFile.getName() + " to list for later processing " );
+                }
+                files.add( repoFileBundle );
+              }
+              zipInputStream.closeEntry();
+              entry = zipInputStream.getNextEntry();
+              continue;
+            } else {
+              getLogger().error( Messages.getInstance().getString( "DefaultImportHandler.ERROR_0011_INVALID_FILE_NAME", sanitizedName ) );
+              throw new PlatformImportException(
+                  Messages.getInstance().getString( "DefaultImportHandler.ERROR_0011_INVALID_FILE_NAME", entryName ), PlatformImportException.PUBLISH_PROHIBITED_SYMBOLS_ERROR );
+            }
```

Comment:
Overall looks good, but this code block is too big and hard to read, I suggest to improve it by breaking it down into smaller portions.

Do you think that it would be beneficial to add some unit tests for this new logic? Breaking down into smaller portions would also help to add UTs.